### DISCUSSION TITLE
is there a way to set the configuration switches in software?

### DISCUSSION THREAD

**Discussion by cm68** [thehighnibble/thehighnibble.github.io#67] [Q&A]
I have my imsai on a shelf, and want to set the memory maps/cpu speed, etc, somewhat frequently.

getting up to push the little reset button on the esp32 module means I need to take it down, hold the examine switch, etc.
this seems quite fiddly.  having this as a GUI option would be very nice.

--curt

**Comment by GrantMeStrength** [comment]
Do you need something different than what this button does?
 

![Screenshot 2023-05-25 at 9 12 26 AM](https://github.com/thehighnibble/thehighnibble.github.io/assets/1928855/7bf3ddd6-f6fb-4e14-a7fd-f5573a1ae5a2)

**Comment by cm68** [comment]
very much so.   that button reboots the simulation, but what I want to do is set the configuration switches
which are documented in https://thehighnibble.com/cromemcoZ1/configure/#startup-configuration-non-volatile-storage-nvs
or the imsai equivalent.

things like, setting the memory map ID, which i think there is no way to do by editing the boot.conf or any other file.

**Comment by dmcnaugh** [comment]
Short answer to your original post: No.

I have marked this as an Enhancement request. However, on my list of priorities is it very low as there is already a way to set configuration via the startup configuration process, and so while convenient, providing an alternate method is not important.
